### PR TITLE
[RDY] Correct log levels

### PR DESCRIPTION
**Pull request by mhanberg**
The log levels declared in the log module were inconsistent with the
MessageType object declared in the protocol module.

This surfaced as the inability to correctly log when using the
nvim-lspconfig package.

This change synchronizes the two logging levels so that the semantics
match, but does not make the naming consistent, as that would result in
a bigger change.

### CODE REVIEW

**Comment by mhanberg** [comment]
@tjdevries @janlazo any reason this can't be merged?

(Sorry, i'm not sure who to ping for this)

**Review by janlazo — APPROVED** [review]

File: runtime/lua/vim/lsp/log.lua [hunk 1]
```diff
@@ -12,10 +12,10 @@ local log = {}
 -- Level numbers begin with 'trace' at 0
 log.levels = {
   TRACE = 0;
```
Comment (janlazo) [inline runtime/lua/vim/lsp/log.lua:14]: `Levels by name` does not match the new order of log levels. Trace should be the last entry of `log.levels`.

Need a test, or a ticket, to demonstrate why these changes are needed. I'm not sure if both modules should be in sync at all.

Comment (mhanberg) [reply runtime/lua/vim/lsp/log.lua:14]: I think I am now understanding how this works now.

I think my misunderstanding draws from the fact that you must set the global log level to `TRACE` if you want to be able to individually control the logging levels from nvim-lspconfig configs, and that these levels are reversed.

`log.levels` has the most verbose logging level set to 0 (TRACE), whereas `protocol.MessageType` has the most verbose logging level set to 4 (Log).

**Comment by mhanberg** [comment]
This change isn't needed, sorry for the noise 😓.

File: runtime/lua/vim/lsp/log.lua (hunk 1 above)
Comment (mhanberg) [reply runtime/lua/vim/lsp/log.lua:14]: I should probably open an issue on the nvim-lspconfig repo sharing that you need to set the global log_level for it to work properly.
